pkg/blockio/blockio.go

```diff
@@ -0,0 +1,60 @@
+//go:build linux
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package blockio
+
+import (
+	"github.com/containerd/containerd/log"
+	"github.com/intel/goresctrl/pkg/blockio"
+	"github.com/pkg/errors"
+
+	"github.com/containerd/containerd/plugin"
+)
+
+// Config contains the configuration of the blockio plugin
+type Config struct {
+	ConfigFile string `toml:"config_file" json:"configFile"`
+}
+
+func init() {
```

Comment:
see comments and possible pattern for implementation over here given the similarity of the pattern.. https://github.com/containerd/containerd/pull/5439#discussion_r714992494

I would suggest you move your init to the cri plugin's init, add a getter for an enabled flag based on config presence logic, possibly add a config flag for ignoring the situation where the annotation is present but the block feature isn't configured.. if you have questions pls ping! Cheers, Mike

Reply:
@mikebrow , thanks for your comments!

As the blockio and RDT patch sets are very similar in the containerd perspective, I thought it would be best to wait and see the final, approvable status of the RDT patchset, and then do the same changes in the blockio patch set. I hope this will save your and other reviewer's time.